admins/README.md

```diff
@@ -0,0 +1,157 @@
+
+We currently run a single zuul launcher on the zuul host, but this component is horizontally-scalable and new instances should be run on separate small hosts. Once a zuul launcher connects to the zuul server via gearman, it will immediately begin handling requests.
+
+Since the zuul server does not control which launcher will run a specific job, all launchers must have access to all test slaves.
+
+#### Zuul Web App
+
+The zuul web app is a very basic web server that serves /status.json, a json object that describes the current state of all pipelines. You can use the script zuul-changes.py from zuul's tools/ directory to translate the json object into shell commands that will restore the current state of the zuul pipelines -- useful for restarting the zuul server, which loses pipeline state.
+
+### Nodepool
+
+Nodepool manages a dynamic pool of Openstack VMs. It can be configured with multiple image types, label types, and VM providers. The images are built using diskimage-builder. A label may contain more than one image (referred to as subnodes), and a separate resource pool is maintained for each label. A provider is an Openstack cloud capable of booting the specified label(s). For more information, refer to [nodepool's documentation](http://docs.openstack.org/infra/nodepool/).
+
+#### Nodepool Server
+
+By default the nodepool server handles all aspects of managing the resources; but the builder, launcher, and deleter can be run in separate processes rather than separate threads for scaling and performance. Regardless, the nodepool server will determine node demand, and will control when to boot nodes, delete nodes, and build images. The demand is determined by connecting to zuul via gearman.
+
+Nodepool receives job-completion messages via zeromq -- a relic left over from using Jenkins with a zeromq plugin. Not all jobs send the zeromq event to trigger node deletion, and zuul's default is to not send the message, but we have enabled it globally in the parametr-function python script.
+
+#### Nodepool Builder
+
+The builder uses diskimage-builder to create images and upload them to glance.
+
+#### Nodepool Launcher
+
+The launcher boots VMs and registers them as available test slaves.
+
+#### Nodepool Deleter
+
+The deleter deletes VMs after they have finished running their assigned job.
+
+### Logs Host
+
+A public web server serving log files as static content. All jobs should be configured to use a publisher to scp test output to this host. After test completion zuul will set the GitHub status url to the parent directory for all jobs for the test run, and on test failure zuul will also comment with a link to the log path for the failing job(s). Anyone on the internet can access the logs after they are published.
+
+We have not yet determined a retention policy.
+
+## Architecture
+
+### Connection Diagrams
+
+#### Zuul
```

Comment:
This can be fixed by changing the header to `Zuul Connection Diagram` or similar. markdownlint is complaining that there is another header on line 29 with an identical title.